### DISCUSSION TITLE
"一" "ー"..?

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [MasterInQuestion/talk#25] [Linguistic]
From: [ https://github.com/liriliri/eruda/discussions/326#Note](https://github.com/liriliri/eruda/discussions/326#Note)

    How to disambiguate "一" (numeral 1) from "ー" (Katakana-Hiragana Prolonged Sound Mark)?
    What to justify the existence of "ー"..?

    See also:
    https://en.wiktionary.org/wiki/ー
    https://util.unicode.org/UnicodeJsps/character.jsp?a=%E3%83%BC
    https://unicode.org/charts/nameslist/n_30A0.html#30FC

**Comment by Adhjie** [comment]
The short answer is: in your referenced Wiktionary entry, the etymology section explained your "why" question.

    Additional etymologies could be found in NKD (Nihon Kokugo Daijiten <日本国語大辞典>) version 3 book form.
    Or digital form in [JapanKnowledge](https://japanknowledge.com/), ["kotobank.jp"](https://kotobank.jp/) (NKD concise edition), ["gogen-yurai.jp"](https://gogen-yurai.jp/) (語源由来辞典) website.
    And other language forums like Cave of Linguists, Wiktionary Discord server.
    .
    I haven't tried other forums because making accounts are tedious. (gotta backup all the recovery options for each new acc)
    Anyway, [WordReference](https://www.wordreference.com/), Stack Exchange resources pages, [kWhazit Oddities](https://kwhazit.ucoz.net/), [WaniKani](https://www.wanikani.com/), [Tofugu](https://www.tofugu.com/), and many many more.

    To answer the "how" question, imagine kanji (漢字) and kana (仮名) in Japanese could be presented with a monospace font like in programming.
    Or using an app to detect any Unicode points: e.g. UnicodePad in Android, BabelPad for Windows.
    There's probably online tool too for this, like online keyboard from Lexilogos.
    .
    Then it become clear they're different: different point in Unicode, visibly different using Unicode detection app.
    But of course hard to discern with just the naked eyes: different usages, different writing system, e.g. kanji vs. kana.

    That's about it, really.
    Just chiming in, since I know Japanese and felt like answering.

    Edit: there are also other resources from NINJAL (National Institute for Japanese Language and Linguistics) websites.

**Reply by MasterInQuestion** [reply]
Well how would you explain... this AΑА Capital Extraordinariness?
    [ https://github.com/MasterInQuestion/talk/discussions/4](https://github.com/MasterInQuestion/talk/discussions/4)

    Whoever expected that ["getElementById"](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById) could be written in 24+ distinct hardly distinguishable ways..?
\
\
    Anyway, thanks for your assistance.
    I'd like to know: if it's acceptable to replace alike (Katakana-Hiragana Prolonged Sound Mark, specifically) with "-"?
    I don't enough understand Japanese.

    Thanks in advance.

**Comment by Adhjie** [comment]
Just copy-paste the Unicode point from the Unicode point entry from any website that you mentioned.
    Or from official Unicode website.

    There are also Unicode Table in Wiktionary, UnicodePad or BabelPad are for checking the Unicode point that you're currently copying in your clipboard:
    https://en.wiktionary.org/wiki/Appendix:Unicode/Katakana
    https://play.google.com/store/apps/details?id=jp.ddo.hotmist.unicodepad
    https://babelstone.co.uk/Fonts/Ogham.html#Input

    You could also type "coffee" in a Japanese IME/keyboard. In Windows there are 2 inputs: hiragana (平仮名) or katakana (片仮名).
    Although yes, after one types in hiragana, and hit 【Space】: the drop-down menu will suggest the katakana version.

    In Mobile the katakana version is in the suggested field: the same place as auto-correction field in English keyboard.
    E.g. Gboard (currently no OS Japanese IME/keyboard app)
[[
Ko (the hyphen-like symbol right of 【L】) Hi
こーひ -> コーヒー
]]
(this is an example using Gboard in Android)
    Whichever you prefer, or easier to do.

    Oh, yeah. Forgot to answer the "E" part.
    UnicodePad usually recommends an alternative to the Latin Capital Letter E or any character: these are usually similar Unicode points for different purpose.
    I think the official Unicode website has more info on that.
    Just try using UnicodePad for all those variants of "E" after you got their official Unicode point name, check their usages.

    Off-Topic:
    There's also a plugin in Aliucord and maybe in BetterDiscord too:
    To normalize fancy usernames of Discord users, well technically that's to convert it.
    UnicodePad is for checking typed characters.

    Oh yeah, the empty YouTube comments also utilize some Unicode points.
    And when those comments are copy-pasted into something like UnicodePad: you could check the Unicode point being used.

    Edit: smh, sorry. I should've continued from the reply. I post it on the wrong section, excuse me. I'm typing from mobile.

**Reply by MasterInQuestion** [reply]
No matter.
    I shall make further refactoring later.

    ----

    You seem to have forgot the entire question...
    Why bother such complexity, after all?

    And how would Japanese tell if it's "一" or "ー"?

**Reply by Adhjie** [reply]
By context, just like any language.
Also, yes. I was being roundabout, also said a wrong thing.
Revision:
Etymology only answered the 'how it came to be' question.
The question of 'what does it mean' is answered by the entry definition in a Japanese dictionary.
'Why' because it's a character used in a language, and language usage is for communication, so as long as it conveys information, it fulfills its purpose.

Katakana long vowel character is always used with words that are represented by katakana.
Those dashes  fall into punctuation category. This is a quick list by tofugu:
Edit: (gotta save the comment first in mobile, there's no auto-save like in email for githuh in mobile browser)
https://www.tofugu.com/japanese/japanese-punctuation

If you check more wiki for this, there are a lot more.
Same thing how Japanese differentiates so many homophones on the spoken language, though; kanji is used for writing because of its visual cue, cultural aspects, and many more.

Edit:
There's also a video by cure dolly, she approach Japanese language through a linguistic POV. No longer upload videos, though.
https://www.youtube.com/watch?v=LDQlGnqElTU

**Reply by Adhjie** [reply]
Oh, there's even a relevant answer from tofugu page:
Japanese:
    〜 波線 (なみせん) or 波ダッシュ (なみだっしゅ)
English:
    〜 Wave Dash

The wave dash isn't really similar to the Western (straight) dash in use. But it's likely the wave dash became popular because straight-line-dashes are already used in katakana to show a long vowel, and not differentiating it here would be confusing.
